erizo/src/erizo/thread/Scheduler.cpp

```diff
@@ -29,8 +29,10 @@ void Scheduler::serviceQueue() {
         new_task_scheduled_.wait(lock);
       }
 
+      current_timeout_ = task_queue_.begin()->first;
       while (!stop_requested_ && !task_queue_.empty() &&
-             new_task_scheduled_.wait_until(lock, task_queue_.begin()->first) != std::cv_status::timeout) {
+             new_task_scheduled_.wait_until(lock, current_timeout_) != std::cv_status::timeout) {
```

Comment:
we're not updating `current_timeout_` within the loop, could it be an issue? what if someone adds an earlier task in the scheduler?

Reply:
yes, that might be an issue. I don't know how we can correctly update the timeout for all threads when a new task is added. Maybe re-setting in the loop and do a `notify_all` on insert?